src/main/java/co/rsk/bitcoinj/core/TransactionWitness.java

```diff
@@ -66,6 +68,105 @@ public byte[] getScriptBytes() {
             return pushes.get(pushes.size() - 1);
     }
 
+    /**
+     replicated logic from {@link Script#getSigInsertionIndex}
+     * */
+    public int getSigInsertionIndex(Sha256Hash sigHash, BtcECKey signingKey) {
+        int witnessSize = getPushCount();
+        int redeemScriptIndex = witnessSize - 1;
+        byte[] redeemScriptData = getPush(redeemScriptIndex);
+        Script redeemScript = new Script(redeemScriptData);
+        RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(redeemScript.getChunks());
+
+        int sigInsertionIndex = 0;
+        int keyIndexInRedeem = redeemScriptParser.findKeyInRedeem(signingKey);
+
+        byte[] emptyByte = new byte[]{};
+        // the pushes that should have the signatures
+        // are between first one (empty byte for checkmultisig bug)
+        // and second to last one (op_notif + redeem script)
+        for (int i = 1; i < getPushCount() - 1; i ++) {
+            byte[] push = getPush(i);
```

Comment:
Didn't it need to ignore the `OP_0` chunk, as it is done for the `Script` class? I guess for `pushes` it would be an empty array, isn't it?

Reply:
Then, shouldn't it ignore empty arrays as it is done in the `Script` class?

Reply:
You mean this https://github.com/rsksmart/bitcoinj-thin/blob/14dd49918279d667f3ad8d5a3b5c728bd7e6bd91/src/main/java/co/rsk/bitcoinj/core/TransactionWitness.java#L91 ?